crates/analyzer/src/traversal/expressions.rs

```diff
@@ -568,8 +570,11 @@ fn tuple_item_index(item: &str) -> Option<usize> {
 
 fn expr_attribute_self(
     scope: &mut BlockScope,
+    value: &Node<fe::Expr>,
     attr: &Node<String>,
 ) -> Result<ExpressionAttributes, FatalError> {
+    check_for_self_param(scope, value.span)?;
```

Comment:
According to my comment a few lines further down, I think we could call with without `?` and simply not return anything (to keep collecting errors).